core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -384,6 +384,12 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
                 enableSslNative = properties.getProperty("quarkus.ssl.native") != null
                         ? Boolean.parseBoolean(properties.getProperty("quarkus.ssl.native"))
                         : false;
+
+                if (!addAllCharsets) {
+                    addAllCharsets = properties.getProperty("quarkus.addAllCharsets") != null
+                            ? Boolean.parseBoolean(properties.getProperty("quarkus.addAllCharsets"))
```

Comment:
it should be `quarkus.add-all-charsets`. We use dashes for config properties.

I think I would have had the exact same approach than for JNI and have a build item. It would just be an empty shell but it would be easier for the consumers.